### PR TITLE
remove duplicate appears in

### LINKED ISSUE #3495: Duplicate Appears In When Problem Appears Twice in a Module
https://usaco.guide/problems/cses-1753-string-matching/solution

Screen Shot 2023-03-04 at 9 48 09 PM

Also in problem search:

Screen Shot 2023-03-04 at 10 44 00 PM

### PR DESCRIPTION
**Pull request by bqi343**
closes #3495

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: src/utils/algolia-queries.ts [hunk 1]
```diff
@@ -135,7 +135,22 @@ const queries = [
...
+            const removeDuplicates = arrayOfObjects => {
+              return arrayOfObjects.filter(
+                (object, index) =>
+                  index ===
+                  arrayOfObjects.findIndex(
+                    obj => JSON.stringify(obj) === JSON.stringify(object)
+                  )
+              );
+            };
+            existingProblem.problemModules = removeDuplicates([
```
Comment (bqi343) [inline src/utils/algolia-queries.ts:149]: for some reason, this didn't have any effect locally. is this expected? @thecodingwizard

Comment (thecodingwizard) [reply src/utils/algolia-queries.ts:149]: Yes. The process to update algolia is separate (and I forgot how... I think it updates on `gatsby build` if certain environment variables are set. See https://github.com/algolia/gatsby-plugin-algolia and https://github.com/cpinitiative/usaco-guide/blob/master/gatsby-config.ts#L116 . There should be a fast way to test this but I forgot...)

Comment (thecodingwizard) [reply src/utils/algolia-queries.ts:140]: I think a more efficient way to do this is to check if `moduleInfo` exists in `existingProblem.problemModules` already; if so, do nothing, if not, add. Can probably be done with something like `existingProblem.problemModules.find(module => module.id === moduleInfo.id)` or something (`find` returns `undefined` if no match is found)

**Review by thecodingwizard — APPROVED** [review]
(I didn't test)

**Comment by bqi343** [comment]
looks like it worked

**Pull request merged**